File: codility/lessons/prefixSum/CountDiv.scala
package codility.lessons.prefixSum

/**
 * Level : painless
 */
object CountDivScala {
    object Solution {
        def solution(a: Int, b: Int, k: Int): Int = countDivisors(b, k) - countDivisors(a, k) + (if (a % k == 0) 1 else 0)
        private def countDivisors(n: Int, k: Int) = n / k + 1
    }
}

File: src/main/java/codility/lessons/prefixSum/CountDiv.java
package codility.lessons.prefixSum;

/**
 *  Level : painless
 */
class CountDiv {
    public int solution(int A, int B, int K) {
        while (A % K != 0 && A < B) A++;
        return (B - A)/K + (A % K == 0 ? 1 : 0);
    }
}
